src/app/brukerprofil/endrenavn/EndreNavnForm.tsx

```diff
@@ -0,0 +1,298 @@
+        this.props.resetEndreNavnReducer();
+        event.preventDefault();
+    }
+
+    validerFornavn() {
+        try {
+            validerNavn(this.state.fornavn.input);
+        } catch (error) {
+            this.setState({
+                fornavn: {
+                    ...this.state.fornavn,
+                    feilmelding: error.message
+                }
+            });
+            throw error;
+        }
+    }
+
+    validerEtternavn() {
+        try {
+            validerNavn(this.state.etternavn.input);
+        } catch (error) {
+            this.setState({
+                etternavn: {
+                    ...this.state.etternavn,
+                    feilmelding: error.message
+                }
+            });
+            throw error;
+        }
+    }
+
+    validerInput() {
+        let validInput = true;
+
+        try {
+            this.validerFornavn();
+        } catch {
+            validInput = false;
+        }
+
+        try {
+            this.validerEtternavn();
+        } catch {
+            validInput = false;
+        }
+
+        return validInput;
+    }
+
+    handleSubmit(event: FormEvent<HTMLFormElement>) {
+        event.preventDefault();
+
+        if (!this.validerInput()) {
+            return;
+        }
+
+        this.setState({
+            formErEndret: false
+        });
+
+        this.props.endreNavn({
+            fødselsnummer: this.props.person.fødselsnummer,
+            fornavn: this.state.fornavn.input,
+            mellomnavn: this.state.mellomnavn.input,
+            etternavn: this.state.etternavn.input
+        });
+    }
+
+    render() {
+        const kanEndreNavn = veilederHarPåkrevdRolle(this.props.veilederRoller) &&
+            brukersNavnKanEndres(this.props.person);
+        return (
+            <form onSubmit={this.handleSubmit}>
+                <Undertittel>Navn</Undertittel>
+                <Infomelding person={this.props.person} veilderRoller={this.props.veilederRoller}/>
+                <NavnInput
+                    label="Fornavn"
+                    state={this.state.fornavn}
+                    onChange={this.fornavnInputChange}
+                    disabled={!kanEndreNavn}
+                />
+                <NavnInput
+                    label="Mellomnavn"
+                    state={this.state.mellomnavn}
+                    onChange={this.mellomnavnInputChange}
+                    disabled={!kanEndreNavn}
+                />
+                <NavnInput
+                    label="Etternavn"
+                    state={this.state.etternavn}
+                    onChange={this.etternavnInputChange}
+                    disabled={!kanEndreNavn}
+                />
+                <FormKnapperWrapper>
+                    <KnappBase
+                        type="standard"
+                        onClick={this.tilbakestillForm}
+                        disabled={!kanEndreNavn || !this.state.formErEndret || !this.navnErEndret()}
```

Comment:
trenger du både formEndret og navnErEndret?

Reply:
Ja, det er litt subtilt men det er nødvendig for å holde styr på staten rundt submittingen, mener jeg å huske.

Når man submitter en endring blir `formEndret` satt til false slik at knappen blir disablet. Hvis vi ikke hadde hatt det, så ville den vært enablet siden navnet er endret i forhold til propsene som blir sendt inn.